All the OpenSSL API used by sngrep has long been available in LibreSSL
except for EVP_CIPHER_CTX_get_cipher_data() which was added in 3.5.0.

Index: src/capture_openssl.h
--- src/capture_openssl.h.orig
+++ src/capture_openssl.h
@@ -66,14 +66,7 @@
 #define OLD_OPENSSL_VERSION 1
 #endif
 
-/* LibreSSL declares OPENSSL_VERSION_NUMBER == 2.0 but does not include most
- * changes from OpenSSL >= 1.1 (new functions, macros, deprecations, ...)
- */
-#if defined(LIBRESSL_VERSION_NUMBER)
-#define MODSSL_USE_OPENSSL_PRE_1_1_API (1)
-#else
 #define MODSSL_USE_OPENSSL_PRE_1_1_API (OPENSSL_VERSION_NUMBER < 0x10100000L)
-#endif
 
 //! Three bytes unsigned integer
 typedef struct uint16 {
